lib/server.js

```diff
@@ -415,6 +388,23 @@ class Server extends EventEmitter {
 
           self.state = 'stopped';
         });
+
+        // Periodically attempt to connect to server and return if successful
```

Comment:
Sorry to do this, but is it possible to separate this PR between listening for the known log messages (`'waiting for connections'`, etc), and refactoring `waitForAvailable` and `checkAvailable`? I think the changes to those functions are uncontroversial and could be merged immediately.

Removing checks for the log messages, however, is potentially a breaking behavioral change. I'm not quite sure what the implications are, so it behooves us to spend a little more time looking into the change.

Reply:
It's no problem, I can do that.

I understand your concern in regard to searching for the 'waiting for connections'. However, this is the one breaking change that is stopping me from implementing `mongodb-topology-manager` in `mongodb-runner` because there is typically a `logpath` given.

Also, I think searching for the 'waiting for connections' message is a bit of a hack, but I suppose polling the server at regular intervals is a bit hacky as well.